diff --git a/meson.build b/meson.build
index 4358deb..7954a7b 100644
--- a/meson.build
+++ b/meson.build
@@ -9,7 +9,7 @@ sysconfdir = join_paths(prefix, get_option('sysconfdir'))
 autostartdir = join_paths(sysconfdir, 'xdg', 'autostart')
 
 rofiDependency = dependency('rofi')
-pluginsdir = rofiDependency.get_variable('pluginsdir')
+pluginsdir = join_paths(libdir, 'rofi')
 
 
 compiler = meson.get_compiler('c')